file-events/src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -66,8 +135,25 @@ WatchPoint::~WatchPoint() {
         SleepEx(0, true);
         close();
     } catch (const exception& ex) {
-        logToJava(LogLevel::WARNING, "Couldn't cancel watch point %s: %s", utf16ToUtf8String(path).c_str(), ex.what());
+        logToJava(LogLevel::WARNING, "Couldn't cancel watch point %s: %s", wideToUtf8String(registeredPath).c_str(), ex.what());
+    }
+}
+
+// Allocate maximum path length
+#define PATH_BUFFER_SIZE 32768
+
+wstring WatchPoint::getPath() {
+    vector<wchar_t> buffer;
+    buffer.reserve(PATH_BUFFER_SIZE);
```

Comment:
Does it matter that we reserve a new buffer each time we call this method or would it be better to reserve the buffer once for the server object and re-use it?

Reply:
I first created the buffer for the server object, but them moved the method to the `WatchPoint` object, and I think this is just fine. It does not have to be super-efficient, and it's not like we are increasing GC time with unnecessary allocations. It's also worth keeping in mind that these operations are a result of (slow) file operations, so it's okay not to process them with the highest possible speed.